lib/merkle-tree.js

```diff
@@ -1224,7 +1224,15 @@ function getStoredNode (storage, index, cache, error) {
       }
 
       const node = { index, size, hash }
-      if (cache !== null) cache.set(index, node)
+
+      if (cache !== null) {
+        const hashCopy = b4a.alloc(hash.byteLength)
```

Comment:
add a one line comment about why we do the copy also for future hans and mathias